### ISSUE TITLE
Rebase with conflict can't be finished

### ISSUE THREAD

**Issue by OndraTakacs** [ForkIssues/TrackerWin#849]
After resolving conflicts the commit fails with this message:

> Done 1 entries.
> Todo 0 entries.
> Current entry: Pick  4cbc5caaa11255394234a74b082c6a6ff2eceffb
> Next entry:  
> Ask message for  4cbc5caaa11255394234a74b082c6a6ff2eceffb
> Cant encode: C:\xampp\htdocs\buranteatr\.git\rebase-merge\fork-message-archive: System.IO.FileNotFoundException: Soubor C:\xampp\htdocs\buranteatr\.git\rebase-merge\fork-message-archive nebyl nalezen.
> N�zev souboru: C:\xampp\htdocs\buranteatr\.git\rebase-merge\fork-message-archive
>    v System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
>    v System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
>    v System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
>    v System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize, Boolean checkHost)
>    v System.IO.File.InternalReadAllText(String path, Encoding encoding, Boolean checkHost)
>    v System.IO.File.ReadAllText(String path)
>    v Fork.RI.Program.WriteMessage(String sha, String outputTodoPath)
> 
> Aborting commit due to empty commit message.
> error: could not commit staged changes.

**Comment by DanPristupov** [comment]
Did you start the interactive rebase from Fork?

> C:\xampp\htdocs\buranteatr.git\rebase-merge\fork-message-archive: System.IO.FileNotFoundException: 

When you start IR Fork stores messages in this files, but it looks like it doesn't exist.

**Comment by OndraTakacs** [comment]
No I didn't, I used interactive rebase. Maybe it would solve the issue, it was the first time I used rebase at all. But I think this is a bug, it shouldn't fail like this.

I have solved this using another git client.

**Comment by DanPristupov** [comment]
OK. In this way I need more details on how to reproduce the problem. What exactly did you do?

**Comment by mxmlnglt** [comment]
Hello,

I'm having the same problem, since installing the new Fork for Windows version 1.51 version.

Here's the message I sent through the support windows in Fork:

> With the new version of Fork for Windows (1.51 released on 2 July 2020) I now have a blocking error when rebasing a branch whose only commit begins with a hash (#) :
> 
> Error popup says:
> 
> Done 1 entries.
> Todo 0 entries.
> Current entry: Pick  df6654bb023d26da9a537afc1f0ce97aadc7183b
> Next entry:  
> Ask message for  df6654bb023d26da9a537afc1f0ce97aadc7183b
> Cant encode: C:\dev\src\git\my-project\.git\rebase-merge\fork-message-archive: System.IO.FileNotFoundException: Le fichier 'C:\dev\src\git\my-project\.git\rebase-merge\fork-message-archive' est introuvable.
> Nom de fichier�: 'C:\dev\src\git\my-project\.git\rebase-merge\fork-message-archive'
>    � System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
>    � System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
>    � System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
>    � System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize, Boolean checkHost)
>    � System.IO.File.InternalReadAllText(String path, Encoding encoding, Boolean checkHost)
>    � System.IO.File.ReadAllText(String path)
>    � Fork.RI.Program.WriteMessage(String sha, String outputTodoPath)
> 
> Aborting commit due to empty commit message.
> error: could not commit staged changes.
> 
> I did not have this problem with previous Fork for Windows version 1.50
> 
> Steps to reproduced:
> 1. have a /develop branch
> 2. checkout a new branch from /develop, let's say /new-branch
> 3. on /new-branch modify a file and commit with a message starting with the hash # as leading character like: #issue-nr This is my commit message
> 4. checkout /develop, modify the same file (so it would lead to a conflict) and commit it (still with the hash # as leading character for the commit message)
> 5. checkout /new-branch again, and try to rebase on /develop: you will get a similar error to above.
> 
> Best regards,

Note: this is really annoying because I didn't alter my basic git config (OK I know Fork embeds its own git version) and on our project, we tend to start every single commit message with the hash character # immediately followed by a numeric value which is the ID of an issue in your bug tracker (Gitlab).

My current workaround follows this answer: https://stackoverflow.com/a/2788524 because in itself, even using `git rebase --continue` on Git Bash fails: it lets your favorite text editor open this file .git/COMMIT_EDIMSG with your commit message on 1st row, but as the commit message starts with the hash character, git seems to consider it a comment; indeed that's what it tells you after you save/close your text editor:
> Aborting commit due to empty commit message.
> error: could not commit staged changes.

**Comment by DanPristupov** [comment]
> on /new-branch modify a file and commit with a message starting with the hash # as leading character like: #issue-nr This is my commit message

I think you need to change the comment character in git which is `#` by default.

https://github.com/ForkIssues/TrackerWin/issues/629#issuecomment-578837694

**Comment by mxmlnglt** [comment]
I updated my above comment.

> > on /new-branch modify a file and commit with a message starting with the hash # as leading character like: #issue-nr This is my commit message
> 
> I think you need to change the comment character in git which is `#` by default.
> 
> [#629 (comment)](https://github.com/ForkIssues/TrackerWin/issues/629#issuecomment-578837694)

In fact, I never had to do this before. Why all of a sudden is this "tweak" necessary?
Hint: I use Fork for more than a year now...

**Comment by DanPristupov** [comment]
I don't know how it is possible. This problem has always been existing I couldn't find a solution for it.

**Comment by mxmlnglt** [comment]
Is there a way to install previous Fork version in order to test (again)?!

**Comment by DanPristupov** [comment]
You probably can find it in `%localappdata%\Fork\Fork-1.0.50` *[updated]*

**Comment by mxmlnglt** [comment]
> You probably can find it in `%localappdata%\Fork\Fork-1.0.50`

You mean `%localappdata%\Fork\app-1.50.0`? Because that's all I've got.

Oh by the way, all previous versions are located in those `Fork.exe_Url_xxx` folders:
**EDIT** nope, there aren't any previous versions, per folder it's just an `AppCenter.config` file plus a `user.config` file inside a folder whose name is a previous Fork version.

![image](https://user-images.githubusercontent.com/29626929/86928101-68dd1180-c134-11ea-9f40-aa3f29f35a69.png)

**Comment by DanPristupov** [comment]
Yes, 1.50 is the previous version.

I just checked previous git and previous Fork and got the same error.

**Comment by mxmlnglt** [comment]
OK, failed...

When I started the 1.50.0 version (through the proper Windows shortcut with "Start Folder" set to point to `C:\Users\<my-windows-user>\AppData\Local\Fork\app-1.50.0`) Fork tried nonetheless to upgrade the embedded Git instance...

Then I tried to alter some props in the `settings.json`; alas I didn't change the correct ones, SO I started the latest installed version of Fork I had at that moment (the 1.51.1) BUT it updated itself... to 1.51.2!!!

Now the folder 1.50.0 has disappeared.

So I'm stuck with both 1.5x "installed" versions...

Only `C:\Users\<my-windows-user>\AppData\Local\Fork\gitInstance` retains the older git version (2.20.1 - installed in late October 2019).

**Comment by mxmlnglt** [comment]
> I just checked previous git and previous Fork and got the same error.

OK, weird then.

I'll just have to set the commentchar as you explained above.

**Comment by psylenced** [comment]
Received this error today:

```
Current entry: Pick  9bxxxxxxxxxxxxxxxxx
Next entry: Pick  89xxxxxxxxxxxxxxxxx
Ask message for  9bxxxxxxxxxxxxxxxxx
Cant encode: F:\Projects\Work\.git\rebase-merge\fork-message-archive: System.IO.FileNotFoundException: Could not find file 'F:\Projects\Work\.git\rebase-merge\fork-message-archive'.
File name: 'F:\Projects\Work\.git\rebase-merge\fork-message-archive'
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize, Boolean checkHost)
   at System.IO.File.InternalReadAllText(String path, Encoding encoding, Boolean checkHost)
   at System.IO.File.ReadAllText(String path)
   at Fork.RI.Program.WriteMessage(String sha, String outputTodoPath)
[detached HEAD 9cxxxxxxxxxxxxxxxxx] Log message here
 15 files changed, 120 insertions(+), 91 deletions(-)
```

Fork version 1.72.3.0

I was doing a (non-interactive) rebase and the error occurred after fixing a conflict and resuming.

**Comment by TanyaPristupova** [comment]
What git instance do you use? Fork uses 2.33.1 by default. Did you switch Fork to some other git instance?

**Comment by TanyaPristupova** [comment]
@psylenced thank you for reporting the problem. We've reproduce it with git 2.36.1. We are handling it right now.

**Comment by TanyaPristupova** [comment]
@psylenced we just released the Fork 1.76 update which fixes the problem.

**Comment by psylenced** [comment]
Thanks for the very prompt response!
